src/native.js

```diff
@@ -280,3 +411,299 @@ function getNativeKeys(adUnit) {
+        len: asset.len || 140
+      }
+    // all extensions to the native bid request are passed as is
+    } else if (key === 'ext') {
+      ortbAsset.ext = {
+        asset
+      };
+      // in `ext` case, required field is not needed
+      delete ortbAsset.required;
+    }
+
+    ortb.assets.push(ortbAsset);
+  }
+  return ortb;
+}
+
+/**
+ * This function converts an OpenRTB native request object to Prebid proprietary
+ * format. The purpose of this function is to help adapters to handle the
+ * transition phase where publishers may be using OpenRTB objects but the
+ *  bidder does not yet support it.
+ * @param {object} openRTBRequest an OpenRTB v1.2 request object
+ * @returns a Prebid legacy native format request
+ */
+export function fromOrtbNativeRequest(openRTBRequest) {
+  if (!isOpenRTBBidRequestValid(openRTBRequest)) {
+    return;
+  }
+
+  const oldNativeObject = {};
+  for (const asset of openRTBRequest.assets) {
+    if (asset.title) {
+      const title = {
+        required: asset.required ? Boolean(asset.required) : false,
+        len: asset.title.len
+      }
+      oldNativeObject.title = title;
+    } else if (asset.img) {
+      const image = {
+        required: asset.required ? Boolean(asset.required) : false,
+      }
+      if (asset.img.w && asset.img.h) {
+        image.sizes = [asset.img.w, asset.img.h];
+      } else if (asset.img.wmin && asset.img.hmin) {
+        image.aspect_ratios = {
+          min_width: asset.img.wmin,
+          min_height: asset.img.hmin,
+          ratio_width: asset.img.wmin,
+          ratio_height: asset.img.hmin
+        }
+      }
+
+      if (asset.img.type === NATIVE_IMAGE_TYPES.MAIN) {
+        oldNativeObject.image = image;
+      } else {
+        oldNativeObject.icon = image;
+      }
+    } else if (asset.data) {
+      let assetType = Object.keys(NATIVE_ASSET_TYPES).find(k => NATIVE_ASSET_TYPES[k] === asset.data.type);
+      let prebidAssetName = Object.keys(PREBID_NATIVE_DATA_KEYS_TO_ORTB).find(k => PREBID_NATIVE_DATA_KEYS_TO_ORTB[k] === assetType);
+      oldNativeObject[prebidAssetName] = {
+        required: asset.required ? Boolean(asset.required) : false,
+      }
+      if (asset.data.len) {
+        oldNativeObject[prebidAssetName].len = asset.data.len;
+      }
+    }
+    // video was not supported by old prebid assets
+  }
+  return oldNativeObject;
+}
+
+/**
+ * Converts an OpenRTB request to a proprietary Prebid.js format.
+ * The proprietary Prebid format has many limitations and will be dropped in
+ * the future; adapters are encouraged to stop using it in favour of OpenRTB format.
+ * @param {BidRequest[]} bidRequests an array of valid bid requests
+ * @returns an array of valid bid requests where the openRTB bids are converted to proprietary format.
+ */
+export function convertOrtbRequestToProprietaryNative(bidRequests) {
+  if (!bidRequests || !isArray(bidRequests)) return bidRequests;
```

Comment:
because a bunch of adapters are calling this now, the body of this should be guarded with `if(FEATURES.NATIVE) { ... }`